### PR TITLE
feat: generate package-lock for faster install

### PR DESCRIPTION
**Pull request by BrandonOCasey**
Generate a package lock for plugin to save ~70s on initial install, even when deps have changed by selecting other options.

### CODE REVIEW

**Comment by gkatsev** [comment]
Interesting idea. With this, we can even run `npm ci` the very first time (maybe with a `--install` flag) which will make that initial install even faster.

**Comment by BrandonOCasey** [comment]
Yeah we could, we would just have to version check npm, let me see if yeoman can do it

**Comment by misteroneill** [comment]
I think you could simply skip the `npmInstall` call and manually invoke `npm ci` - if I'm understanding everything correctly.

Really cool idea! 👍

File: package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "test": "npm-run-all test:*",
     "test:unit": "mocha test/*.test.js",
     "test:integration": "node scripts/run-integration-tests.js",
-    "preversion": "npm run test",
+    "preversion": "npm-run-all generate-pkg-lock test",
+    "generate-pkg-lock": "node scripts/generate-pkg-lock.t js && git add generators/app/templates/_package-lock.json",
```
Comment (BrandonOCasey) [inline package.json]: generate package lock and add it every version, I think this makes sense to me.

File: scripts/run-integration-tests.js [hunk 2]
```diff
@@ -12,8 +12,9 @@ let tempDir;
 helpers.run(libs.GENERATOR_PATH)
   .inTmpDir(function(dir) {
     tempDir = dir;
+    console.log(`Generating Project in ${tempDir}`);
   })
-  .withOptions(libs.options())
+  .withOptions(Object.assign(libs.options(), {skipInstall: false}))
```
Comment (BrandonOCasey) [inline scripts/run-integration-tests.js]: I removed the install steps below, instead we do the builtin yeoman install

**Pull request merged**